services/Authentication/src/Authentication.BusinessLogicLayer/Services/Implementations/AuthenticationService.cs

```diff
@@ -0,0 +1,185 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using Authentication.BusinessLogicLayer.DataTransferObjects;
+using Authentication.BusinessLogicLayer.Services.Interfaces;
+using Authentication.DataAccessLayer.Entities.ConfigurationModels;
+using Authentication.DataAccessLayer.Entities.Exceptions.BadRequest;
+using Authentication.DataAccessLayer.Entities.Models;
+using AutoMapper;
+using Common;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Authentication.BusinessLogicLayer.Services.Implementations;
+
+public class AuthenticationService : IAuthenticationService
+{
+    private readonly IMapper _mapper;
+
+    private readonly UserManager<User> _userManager;
+    private readonly RoleManager<IdentityRole> _roleManager;
+
+    private readonly JwtConfiguration _jwtConfiguration;
+    private User? _user;
+
+    public AuthenticationService(IMapper mapper, UserManager<User> userManager, 
+        RoleManager<IdentityRole> roleManager, IOptionsSnapshot<JwtConfiguration> configuration)
+    {
+        _mapper = mapper;
+        _userManager = userManager;
+        _roleManager = roleManager;
+        _jwtConfiguration = configuration.Value;
+    }
+
+
+    public async Task<IdentityResult> RegisterUser(UserForRegistrationDto userForRegistration)
+    {
+        var user = _mapper.Map<User>(userForRegistration);
+        var result = await _userManager.CreateAsync(user, userForRegistration.Password!);
+
+        foreach (var role in userForRegistration.Roles!)
+            if (!await _roleManager.RoleExistsAsync(role))
+            {
+                return IdentityResult.Failed(new IdentityError
+                {
+                    Code = "RoleNotExist",
+                    Description = $"Role {role} does not exist."
+                });
+            }
+
+        if (result.Succeeded)
+            await _userManager.AddToRolesAsync(user, userForRegistration.Roles!);
```

Comment:
The user does not have to enter his role during registration. You assign the user role to the service yourself.